src/main/java/org/swisspush/reststorage/FileSystemStorage.java

```diff
@@ -322,7 +262,7 @@ private void deleteEmptyParentDirs(String path) {
         });
     }
 
-    private String canonicalize(String path) {
+    public String canonicalize(String path) {
```

Comment:
Why is this now public?

Reply:
Because _FileSystemStorage_ now implements _FilePutterCallbacks_.

Alternative 1: We could pass an inline implementation delegating to that method instead at [FileSystemStorage:175](https://github.com/hiddenalpha/vertx-rest-storage/blob/d1629b6852389e36e119b3e50b2124278a39965d/src/main/java/org/swisspush/reststorage/FileSystemStorage.java#L175).

Alternative 2: We can introduce a code duplication by duplicate that method from FileSystemStorage into FilePutter.